README.md

```diff
@@ -10,7 +10,22 @@ If you want the "quick and easy" setup like in CI:
 
 Else: you CAN set CMake options to compile ABC and Yosys from source.
 
+NOTE: this project is Rust first, C++ second but for "ease of dev" it is still structured to
+be compiled using the classic `mkdir build && cd build && cmake ..` which is why there is a `CMakeLists.txt` in the root.
+
+## Rust wrapper: circuit-gen-rs/
+
+Thin Rust wrapper around `lib_circuits` to easily call it from `pallet-ocw-circuits`.
+
+IMPORTANT: this NOT meant to be compiled with eg `cd build && cmake ..` in the root of this repository.
+It is meant to be compiled via `cargo build`
```

Comment:
Precision: eg "it CAN be compiled with cmake .. && .. && ctest" but the tests are REALLY basic. Almost all the testing is done Rust side.